src/Generating/FinishGen.h

```diff
@@ -70,6 +70,75 @@ class cFinishGenNetherClumpFoliage :
 
 
 
+class cFinishGenClumpTopBlock :
+	public cFinishGen
+{
+public:
+	// Contains the meta, type and weight for a clump block
+	struct FoliageInfo
+	{
+		BLOCKTYPE m_BlockType;
+		NIBBLETYPE m_BlockMeta;
+		int m_Weight;
+
+		FoliageInfo(BLOCKTYPE a_BlockType, NIBBLETYPE a_BlockMeta, int a_Weight) :
+			m_BlockType(a_BlockType),
+			m_BlockMeta(a_BlockMeta),
+			m_Weight(a_Weight)
+		{}
+	};
+
+	// Contains the minimum and maximum amount of clumps for a biome and it's blocks.
+	struct BiomeInfo
+	{
+		int m_MinNumClumpsPerChunk;
+		int m_MaxNumClumpsPerChunk;
+		std::vector<FoliageInfo> m_Blocks;
+
+		BiomeInfo() :
+			m_MinNumClumpsPerChunk(0),
+			m_MaxNumClumpsPerChunk(2),
+			m_Blocks()
+		{}
+
+		BiomeInfo(int a_MinNumClumpsPerChunk, int a_MaxNumClumpsPerChunk, std::vector<FoliageInfo> a_Blocks) :
+			m_MinNumClumpsPerChunk(a_MinNumClumpsPerChunk),
+			m_MaxNumClumpsPerChunk(a_MaxNumClumpsPerChunk),
+			m_Blocks(a_Blocks)
+		{}
+	};
+
+
+	cFinishGenClumpTopBlock(int a_Seed, std::vector<BiomeInfo> a_BlockList) :
+		m_Noise(a_Seed),
+		m_FlowersPerBiome()
+	{
+		std::swap(a_BlockList, m_FlowersPerBiome);
+	}
+
+	/** Parses a string and returns a vector with a length of biMaxVariantBiome.
+	The format of the string is "<Biomes separated with a comma>;<Blocks separated with a comma>". This can also be repeated with a | */
+	static void ParseConfigurationString(AString a_String, std::vector<BiomeInfo> & a_Output = std::vector<BiomeInfo>());
```

Comment:
The doxycomment for the format seems wrong, at least compared to the default values inserted in `ParseIniFile()`